Source/DafnyCore/Verifier/Translator.ClassMembers.cs

```diff
@@ -665,27 +674,51 @@ public partial class Translator {
 
         Contract.Assert(definiteAssignmentTrackers.Count == 0);
       } else {
-        // check well-formedness of any default-value expressions (before assuming preconditions)
-        foreach (var formal in m.Ins.Where(formal => formal.DefaultValue != null)) {
-          var e = formal.DefaultValue;
-          CheckWellformed(e, new WFOptions(null, false, false, true), localVariables, builder, etran);
-          builder.Add(new Boogie.AssumeCmd(e.tok, CanCallAssumption(e, etran)));
-          CheckSubrange(e.tok, etran.TrExpr(e), e.Type, formal.Type, builder);
+        var readsCheckDelayer = new ReadsCheckDelayer(etran, null, localVariables, builderInitializationArea, builder);
 
-          if (formal.IsOld) {
-            Boogie.Expr wh = GetWhereClause(e.tok, etran.TrExpr(e), e.Type, etran.Old, ISALLOC, true);
-            if (wh != null) {
-              var desc = new PODesc.IsAllocated("default value", "in the two-state lemma's previous state");
-              builder.Add(Assert(e.tok, wh, desc));
+        // check well-formedness of any default-value expressions (before assuming preconditions)
+        readsCheckDelayer.WithDelayedReadsChecks(true, wfo => {
```

Comment:
AFAIK the prefix `With` is commonly used for pure methods, while this seems to be a side-effecting one. I suggest replacing `With` with `Add`.

Reply:
Good observation, it makes sense

Reply:
Fair point, although I do want to get across the idea of "execute this with this additional context/before and after effect". Perhaps `RunWithDelayedReadsChecks` (or is there a more idiomatic verb for executing an `Action<...>`?)

Reply:
Went with "DoWith.." :)